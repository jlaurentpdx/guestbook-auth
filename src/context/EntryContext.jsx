import { useContext, createContext, useState, useEffect } from 'react';
import { fetchEntries } from '../services/entries';

export const EntriesContext = createContext();

export function EntriesProvider({ children }) {
  const [entries, setEntries] = useState([]);

  useEffect(() => {
    const fetchData = async () => {
      const data = await fetchEntries();
      setEntries(data);
    };
    fetchData();
  }, []);

  return (
    <EntriesContext.Provider value={{ entries, setEntries }}>
      {children}
    </EntriesContext.Provider>
  );
}

export const useEntries = () => {
  const context = useContext(EntriesContext);

  if (context === undefined)
    throw new Error('useEntries can not be used outside of EntriesProvider');

  return context;
};
