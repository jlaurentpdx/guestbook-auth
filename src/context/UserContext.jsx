import { useState, useContext, useMemo, createContext } from 'react';

export const UserContext = createContext();

export function UserProvider({ children }) {
  const [user, setUser] = useState('');

  const value = useMemo(() => ({ user, setUser }), [user]);

  return <UserContext.Provider value={value}>{children}</UserContext.Provider>;
}

export const useUser = () => {
  const context = useContext(UserContext);

  if (context === undefined) {
    throw new Error('useUser can not be used outside of UserProvider');
  }

  return context;
};
