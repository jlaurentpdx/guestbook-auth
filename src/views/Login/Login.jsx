import { useState } from 'react';
import { useUser } from '../../context/UserContext';

export default function Login() {
  const { login } = useUser();
  const [email, setEmail] = useState('');
  const [password, setPassword] = useState('');

  return (
    <form>
      <label htmlFor="email">Email</label>
      <input
        id="email"
        name="email"
        type="email"
        value={email}
        onChange={(e) => setEmail(e.target.value)}
      />
      <label htmlFor="password">Password</label>
      <input
        id="password"
        name="password"
        type="password"
        value={password}
        onChange={(e) => setPassword(e.target.value)}
      />
      <button
        onClick={(e) => {
          e.preventDefault();
          login(email, password);
        }}
      >
        Submit
      </button>
    </form>
  );
}
