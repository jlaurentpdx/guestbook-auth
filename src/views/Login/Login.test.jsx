import { render, screen } from '@testing-library/react';
import { MemoryRouter as Router } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import { UserProvider } from '../../context/UserContext';

import Login from './Login';

test('should render a Login form', () => {
  const container = render(
    <UserProvider>
      <Router>
        <Login />
      </Router>
    </UserProvider>
  );

  //   screen.getByRole('textbox', { name: /password/i }); <-- Not sure why this did not work
  const email = screen.getByRole('textbox', { name: /email/i });
  const password = screen.getByLabelText(/password/i);
  const submit = screen.getByRole('button', { name: /submit/i });

  userEvent.type(email, 'test@example.com');
  userEvent.type(password, 'test1234');

  expect(email).toHaveValue('test@example.com');
  expect(password).toHaveValue('test1234');
  expect(submit).toBeEnabled();

  expect(container).toMatchSnapshot();
});
