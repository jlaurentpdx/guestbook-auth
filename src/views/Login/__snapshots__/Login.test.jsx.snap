// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render a Login form 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="form"
      >
        <div
          class="control"
        >
          <label
            class="label"
            for="email"
          >
            Email
          </label>
          <input
            id="email"
            type="email"
            value="test@example.com"
          />
        </div>
        <div
          class="control"
        >
          <label
            class="label"
            for="password"
          >
            Password
          </label>
          <input
            id="password"
            type="password"
            value="test1234"
          />
        </div>
        <button>
          Submit
        </button>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="form"
    >
      <div
        class="control"
      >
        <label
          class="label"
          for="email"
        >
          Email
        </label>
        <input
          id="email"
          type="email"
          value="test@example.com"
        />
      </div>
      <div
        class="control"
      >
        <label
          class="label"
          for="password"
        >
          Password
        </label>
        <input
          id="password"
          type="password"
          value="test1234"
        />
      </div>
      <button>
        Submit
      </button>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
